Azure/sonic-buildimage
# [device/celestica]: Implement Seastone2 platform API

**- What I did**
  - Implement Seastone2 platform API

**- How I did it**
1. Implement Fan APIs.
2. Implement Psu APIs.
3. Implement Thermal APIs.
4. Implement Component APIs.
5. Implement Chassis APIs.
6. Implement Sfp APIs.
7. Implement Watchdog APIs. (Required new CPLD version)
8. Upgrade switchboard driver to support port insert/remove interrupt
9. Init all above APIs class on chassis API.

**- How to verify it**
- Create python script follow the new platform API design doc to verify it (Azure/SONiC#285.)
- Verify result from SONIC commands that will call sonic_platform APIs
  - [seastone2-api-cli-test.log](https://github.com/Azure/sonic-buildimage/files/5020170/seastone2-api-cli-test.log)


Signed-off-by: Wirut Getbamrung wgetbumr@celestica.com
